src/Utils/imageCaptureUtils.ts

```diff
@@ -0,0 +1,388 @@
+      width = 400;
+      height = Math.round(img.height * (width / img.width));
+    }
+  }
+
+  return {
+    width,
+    height,
+    originalWidth: img.width,
+    originalHeight: img.height,
+  };
+};
+
+/**
+ * Creates a canvas with the image drawn to it with the specified dimensions
+ * @param img - The image element to draw
+ * @param width - The desired width
+ * @param height - The desired height
+ * @returns Canvas element with the image drawn to it, or null if failed
+ */
+export const createImageCanvas = (
+  img: HTMLImageElement,
+  width: number,
+  height: number,
+): HTMLCanvasElement | null => {
+  const canvas = document.createElement("canvas");
+  canvas.width = width;
+  canvas.height = height;
+
+  const ctx = canvas.getContext("2d");
+  if (!ctx) {
+    return null;
+  }
+
+  // Use higher quality rendering
+  ctx.imageSmoothingEnabled = true;
+  ctx.imageSmoothingQuality = "high";
+  ctx.drawImage(img, 0, 0, width, height);
+
+  return canvas;
+};
+
+/**
+ * Creates a scaled-down version of the canvas for size reduction
+ * @param img - The image element to draw
+ * @param width - The original width
+ * @param height - The original height
+ * @param scale - The scale factor (0-1)
+ * @returns A new canvas with reduced dimensions
+ */
+export const createScaledCanvas = (
+  img: HTMLImageElement,
+  width: number,
+  height: number,
+  scale: number = 0.8,
+): HTMLCanvasElement => {
+  const smallerCanvas = document.createElement("canvas");
+  smallerCanvas.width = width * scale;
+  smallerCanvas.height = height * scale;
+
+  const smallerCtx = smallerCanvas.getContext("2d");
+  if (smallerCtx) {
+    smallerCtx.drawImage(img, 0, 0, smallerCanvas.width, smallerCanvas.height);
+  }
+
+  return smallerCanvas;
+};
+
+/**
+ * Converts a canvas to a File object with the appropriate format and compression
+ * @param canvas - The canvas to convert
+ * @param filename - The filename to use
+ * @param format - The image format ('image/png' or 'image/jpeg')
+ * @param quality - The compression quality (0-1)
+ * @returns Promise that resolves to a File object or null if failed
+ */
+export const canvasToFile = (
+  canvas: HTMLCanvasElement,
+  filename: string,
+  format: string = "image/png",
+  quality: number = 0.9,
+): Promise<File | null> => {
+  return new Promise((resolve) => {
+    canvas.toBlob(
+      (blob) => {
+        if (blob) {
+          const file = new File([blob], filename, { type: format });
+          resolve(file);
+        } else {
+          resolve(null);
+        }
+      },
+      format,
+      quality,
+    );
+  });
+};
+
+/**
+ * Processes the captured image with appropriate size and quality constraints
+ * @param imageDataUrl - The data URL of the captured image
+ * @param t - Translation function for error messages
+ * @returns Promise that resolves to a File object or null if processing failed
+ */
+export const processWebcamImage = (
+  imageDataUrl: string,
+  t: (key: string) => string,
```

Comment:
let's not pass `t` function. I'd rather make this a hook instead to be able to use translations.